### PR TITLE
Built-in Function: bin

### PR DESCRIPTION
**Pull request by silentred**
Also fixed three places where golint throws warning.
```
plan/logical_plan_builder.go:851:2: can probably use "var sortMergeTables []model.CIStr" instead
plan/physical_plan_builder.go:587:2: can probably use "var result [][]*requiredProperty" instead
plan/physical_plan_builder.go:588:2: can probably use "var condIndex []int" instead
```

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1572,7 +1572,24 @@ type builtinBinSig struct {
...
+		return d, nil
+	}
+
+	num, err := arg.ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	bits := fmt.Sprintf("%b", uint64(num))
+	d.SetString(bits)
+
```
Comment (shenli) [inline expression/builtin_string.go]: Remove the empty line.

File: expression/builtin_string.go [hunk 2]
```diff
@@ -1572,7 +1572,24 @@ type builtinBinSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.6/en/string-functions.html#function_bin
 func (b *builtinBinSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("bin")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	arg := args[0]
```
Comment (shenli) [inline expression/builtin_string.go]: Check if arg.IsNil()

File: expression/builtin_string_test.go [hunk 3]
```diff
@@ -1163,3 +1163,32 @@ func (s *testEvaluatorSuite) TestElt(c *C) {
...
+		{"10", "1010"},
+		{"10.2", "1010"},
+		{"10aa", "1010"},
+		{"10.2aa", "1010"},
+		{"aaa", "0"},
+		{"", nil},
+		{10, "1010"},
+		{10.0, "1010"},
+		{-1, "1111111111111111111111111111111111111111111111111111111111111111"},
+		{"-1", "1111111111111111111111111111111111111111111111111111111111111111"},
```
Comment (shenli) [inline expression/builtin_string_test.go]: Add a test case for nil input.

**Comment by shenli** [comment]
@silentred Thanks for your PR! Please resolve the conflicts and address the comments.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
@silentred 
Thank you for your contribution!

**Pull request merged**